.github/workflows/test-demo.yaml

```
@@ -0,0 +1,60 @@
+name: Test VM demo
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+  workflow_dispatch:
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: 'actions/checkout@v4'
```

See [zizmor - artipacked](https://woodruffw.github.io/zizmor/audits/#artipacked) audit rule

```suggestion
      - uses: 'actions/checkout@v4'
        with: { persist-credentials: false }
```